Projects/rym/rick and morty/src/App.jsx

```diff
@@ -0,0 +1,52 @@
+import SearchInput from "./components/search-input/search-input"
+import CharactersList from "./components/characters-list/characters-list"
+import getCharacters from "./services/get_characters";
+import { useState,useEffect } from "react";
+
+function useCharacters() {
+  const [characters, setCharacters] = useState([])
+
+  const searchCharacters = (value='') => getCharacters().then(data => {
+        const charactersFilter = data.results.filter(character =>
+          character.name.toLowerCase().includes(value.toLowerCase())
+        )
+        setCharacters(charactersFilter)
+      }).catch(error => {
+          console.error('Error al obtener los personajes:', error)
+    })
+
+    return {
+      searchCharacters,
+      characters
+    }
+}
+
+function App() {
+  const [searchValue, setSearchValue] = useState('')
+  const {searchCharacters,characters} = useCharacters()
+  useEffect(() => {
+    searchCharacters(searchValue)
+  },[searchValue])
+
+  return (
+    <>
+      <img src="http://rym-alvarzrz.s3-website-us-west-1.amazonaws.com/images/background.jpeg" alt="" />
+      <div>
+        <SearchInput searchValue={searchValue} setSearchValue={setSearchValue}/>
+        <ul>
+          {characters.map(characters => (
```

Comment:
You have to ve careful here, you are using the same name in both places,

![image](https://github.com/juan982610/Juan_Rozo/assets/25773197/de0a3146-3113-4adc-a9d6-bdce598e7820)